### ISSUE TITLE
Expose the built-in markdown parser to render markdown-valued variables with

### ISSUE THREAD

**Issue by aentwist** [vuejs/vitepress#2410]
### Is your feature request related to a problem? Please describe.

My CMS stores markdown values in frontmatter data. For example, an intro section `intro` with a markdown string value. Currently, trying to access it in the page with `{{ $frontmatter.intro }}` just renders it as a string (a direct result of the markdown -> Vue -> HTML pipeline). What is the recommended way to call the markdown parser on it?

### Describe the solution you'd like

It would be nice to see this exposed somehow (getting the built-in markdown-it instance, for example).

### Describe alternatives you've considered

I could install markdown-it separately and use it that way?

### Validations

- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.dev).
- [X] Read the [Contributing Guidelines](https://github.com/vuejs/vitepress/blob/main/.github/contributing.md).
- [X] Check that there isn't already an issue that asks for the same feature to avoid creating a duplicate.

**Comment by aentwist** [comment]
Well, it just occurred to me that this use case is probably unusual... so allow me to elaborate.

I want to build a landing page and allow non-technical users to edit specific content (potentially using rich text I guess?) without being exposed to the structure.

My strategy is to use a custom theme that makes use of frontmatter keys to "slot in" content into specific places in the page instead of allowing the whole page to be edited with the CMS's special "body" frontmatter field (that makes up the non-frontmatter, or main, markdown content for pages).

**Comment by Aganivi** [comment]
createMarkdownRenderer don't seem to work？

**Comment by Marhc** [comment]
+1

**Comment by adiroiban** [comment]
Having a public API to get the globlal `markdownit` instance would be nice.

The main code is something like this

```vue
<script setup lang="ts">
import {useData} from 'vitepress'
import markdownit from 'markdown-it'
const {frontmatter} = useData()
const md = markdownit({
  html: true,
  linkify: true,
  typographer: true,
})
</script>

<template>
  <span
    v-html="md.renderInline(frontmatter.topSection.note)"
    class="font-display block text-base font-semibold text-red-500"
  >
</template>
```

As a first step, it woud be nice to have  `import {md} from 'vitepress'` so that we get the markdown instance already configured

Better yet. something like `<vp-md :model="frontmatter.topSection.note" inline class="some class">` should help reduce the boilerplate

**Comment by adiroiban** [comment]
I ended up using something like this... via the transformPageData hook 

I wanted to have markdown render for some of the frontmatter values

I get the general markdown renderer only once... I don't know how else to get it

```
    async transformPageData(pageData, { siteConfig}) {
        if (!global_md) {
            // Create the markdown renderer only once.
            global_md = await createMarkdownRenderer(
                siteConfig.srcDir,
                siteConfig.markdown,
                siteConfig.site.base,
                siteConfig.logger
            )
        }

        // Render inline markdown for frontmatter ending with _md
        Object.keys(pageData.frontmatter).forEach(key => {
            if (!key.endsWith('_md')) {
                return
            }
            pageData.frontmatter[key] = global_md.renderInline(
                pageData.frontmatter[key]
            )
        })
    },
```


It would be nice to have some sort of "official" support, with link validation

**Comment by bluwy** [comment]
`createMarkdownRenderer` is probably the answer here but needs documentation, and is still experimental.

**Comment by brc-dd** [comment]
I'm thinking of adding `$frontmatter` fast path in the renderer itself and exporting a singleton through something like `async renderMd(md: string): Promise<string>` and dropping `createMarkdownRenderer`. That will fix issues like #5162 too. `renderMd` will be Node-side only though because our markdown-it plugins need fs. They can't work on client. For client-side, the best way is to use the data loader or run from `transformPageData` instead. Also, `renderMd` will reject any call before vitepress is actually up -- it'll be usable inside the hooks and loaders.

**Comment by adiroiban** [comment]
If the requirement is strictly: `My CMS stores markdown values in frontmatter data.`
I think that the best experience is to have automatic markdown rendering.

```
---
title: Docs with VitePress
editLink: true
md:
  title: Generate pages with **VitePress**
---
```

Everyting that is inside the `md` frontmatter key is automatically available as HTML


```html
<template>
  <span
    v-html="frontmatter.md.title"
    class="font-display block text-base font-semibold text-red-500"
  >
</template>
```

It might be nice to export the frontmatter as `fm` .
I think it will make the dev experience nicer 


```html
<template>
  <span
    v-html="fm.md.title"
    class="font-display block text-base font-semibold text-red-500"
  >
</template>
```
